float32 px
float32 py
float32 vx
float32 vy

